worlds/yachtdice/__init__.py

```diff
@@ -0,0 +1,459 @@
+                weights[1] = 0  # don't allow >= 6 rolls
+
+            # Don't allow too many multipliers
+            if multipliers_added > 50:
+                weights[2] = 0
+                weights[3] = 0
+
+            # Don't allow too many extra points
+            if extra_points_added > 300:
+                weights[5] = 0
+
+            # if all weights are zero, allow to add fixed score multiplier, double category, points.
+            if sum(weights) == 0:
+                if multipliers_added <= 50:
+                    weights[2] = 1
+                weights[4] = 1
+                if extra_points_added <= 300:
+                    weights[5] = 1
+
+            # Next, add the appropriate item. We'll slightly alter weights to avoid too many of the same item
+            which_item_to_add = self.multiworld.random.choices([0, 1, 2, 3, 4, 5], weights=weights)[0]
+            item_to_add = ""
+            if which_item_to_add == 0:
+                weights[0] /= 1 + frags_per_dice
+                if frags_per_dice == 1:
+                    return "Dice"
+                else:
+                    return "Dice Fragment"
+            elif which_item_to_add == 1:
+                weights[1] /= 1 + frags_per_roll
+                if frags_per_roll == 1:
+                    return "Roll"
+                else:
+                    return "Roll Fragment"
+            elif which_item_to_add == 2:
+                weights[2] /= 1.05
+                multipliers_added += 1
+                return "Fixed Score Multiplier"
+            elif which_item_to_add == 3:
+                weights[3] /= 1.1
+                multipliers_added += 1
+                return "Step Score Multiplier"
+            elif which_item_to_add == 4:
+                cat_weights = [2, 2, 1, 2, 2, 2, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1]
+                weights[4] /= 1.1
+                return self.multiworld.random.choices(possible_categories, weights=cat_weights)[0]
+            elif which_item_to_add == 5:
+                score_dist = self.options.points_size.value
+                probs = [1, 0, 0]
+                if score_dist == 1:
+                    probs = [0.9, 0.08, 0]
+                if score_dist == 2:
+                    probs = [0, 1, 0]
+                if score_dist == 3:
+                    probs = [0, 0.3, 0.7]
+                if score_dist == 4:
+                    probs = [0.3, 0.4, 0.3]
+                c = self.multiworld.random.choices([0, 1, 2], weights=probs)[0]
+                if c == 0:
+                    weights[5] /= 1.01
+                    extra_points_added += 1
+                    return "1 Point"
+                elif c == 1:
+                    weights[5] /= 1.1
+                    extra_points_added += 10
+                    return "10 Points"
+                elif c == 2:
+                    weights[5] /= 2
+                    extra_points_added += 100
+                    return "100 Points"
```

Comment:
Again, I would find this so much nicer as:

```
score_dist = self.options.points_size.value

probs = { "1 Point": 1, "10 Points": 0, "100 Points": 0}
if score_dist == 1:
    probs = { "1 Point": 0.9, "10 Points": 0.08, "100 Points": 0}
... you get the idea ...

choice = self.random.choices(probs.keys(), weights=probs.values())[0]
if choice == "1 Point":
    weights["Extra Points"] /= 1.01
    extra_points_added += 1
elif choice == "10 Points":
    ...

return choice
```

Oh well like it's not the end of the world or anything

*Low priority*